editor/find_in_files_scanner.cpp

```diff
@@ -0,0 +1,352 @@
+			// They get escaped before being written into the file, so things like quotes gets encoded into \".
+			int i = 0;
+			for (auto &scan_match : matches.slice(matches_offset)) {
+				String unescaped_line = scan_match.line.c_unescape();
+				int diffs_found = 0;
+				for (int j = 0; j <= scan_match.end; ++j) {
+					if (scan_match.line[j] == unescaped_line[j - diffs_found]) {
+						continue;
+					}
+
+					diffs_found++;
+				}
+				scan_match.begin -= diffs_found;
+				scan_match.end -= diffs_found;
+				scan_match.line = unescaped_line;
+				matches.insert(matches_offset + i, scan_match);
+				matches.remove_at(matches_offset + i + 1);
+				i++;
+			}
+
+			matches_offset = matches.size();
+		}
+	}
+}
+
+void BuiltinScriptFindInFilesScanner::replace(Ref<FileAccess> file, String file_path, Vector<FindInFilesScanner::ScanLocation> &locations, bool match_case, bool whole_words, String search_text, String new_text) {
+	for (auto &match : _parse_tscn(file)) {
+		if ((file->get_path() + "::" + match.key) != file_path) {
+			continue;
+		}
+
+		file->seek(0);
+		// Account for the edge cases also found during scanning
+		for (auto &location : locations) {
+			if (location.line_number == 1) {
+				location.begin += SCRIPT_SOURCE_TAG_LEN;
+				location.end += SCRIPT_SOURCE_TAG_LEN;
+			}
+		}
+		DefaultFindInFilesScanner::replace(file, file->get_path(), locations, match_case, whole_words, search_text, new_text, match.value.script_start_line - 1);
+	}
+}
+
+HashMap<String, BuiltinScriptFindInFilesScanner::SubResource> BuiltinScriptFindInFilesScanner::_parse_tscn(Ref<FileAccess> file) {
+	enum BuiltinScriptMarkers {
+		SUB_RESOURCE_MARKER,
+		SCRIPT_SOURCE_MARKER,
+		SCRIPT_EOF_MARKER
+	};
+	HashMap<BuiltinScriptMarkers, String> builtin_script_markers_map;
+	builtin_script_markers_map.insert(BuiltinScriptMarkers::SUB_RESOURCE_MARKER, "[sub_resource type=\"GDScript\" id=\"");
+	builtin_script_markers_map.insert(BuiltinScriptMarkers::SCRIPT_SOURCE_MARKER, "script/source = \"");
+	builtin_script_markers_map.insert(BuiltinScriptMarkers::SCRIPT_EOF_MARKER, "\"");
+
+	int script_start_line = -1;
+	String scene_id;
+	String line = file->get_line();
+	BuiltinScriptMarkers looking_for = BuiltinScriptMarkers::SUB_RESOURCE_MARKER;
+	HashMap<String, BuiltinScriptFindInFilesScanner::SubResource> matches;
+	HashMap<int, String> node_names;
+
+	for (int line_idx = 1; !file->eof_reached(); line_idx++, line = file->get_line()) {
+		String pattern = builtin_script_markers_map.get(looking_for);
+		if (!line.begins_with(pattern)) {
+			// Since we're going through the file any ways, let's look for nodes using the builtin scripts found so far, so we can assemble the display_name using the node name
+			if (line.begins_with("[node name=\"")) {
+				// Found a node, store it, alongside it's full name and line index
+				String node_name = line.get_slice("[node name=\"", 1).get_slice("\"", 0);
+				String parent = line.get_slice(" parent=\"", 1).get_slice("\"", 0);
```

Comment:
IMO keeping track of node's parent is not very useful.
Script name would be better, but in most cases the node name is sufficient.